aws/src/test/java/org/apache/iceberg/aws/glue/DynamoLockManagerTest.java

```diff
@@ -0,0 +1,213 @@
+  private static final String TABLE_NAME = "dynamoLockTable";
+
+  private Process dynamoLocalProcess;
+  private DynamoDbClient dynamo;
+  private AwsProperties properties;
+  private LockManager lockManager;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    Runtime rt = Runtime.getRuntime();
+    Process pr = rt.exec(String.format("curl %s -o %s.tar.gz", DYNAMO_LOCAL_DOWNLOAD_URL, DYNAMO_LOCAL_DIR_NAME));
+    pr.waitFor();
+    pr = rt.exec(String.format("mkdir %s", DYNAMO_LOCAL_DIR_NAME));
+    pr.waitFor();
+    pr = rt.exec(String.format("tar xvzf %s.tar.gz -C %s", DYNAMO_LOCAL_DIR_NAME, DYNAMO_LOCAL_DIR_NAME));
+    pr.waitFor();
+  }
+
+  @Before
+  public void before() throws Exception {
+    Runtime rt = Runtime.getRuntime();
+    dynamoLocalProcess = rt.exec(String.format("java -Djava.library.path=%s/DynamoDBLocal_lib -jar " +
+            "%s/DynamoDBLocal.jar -inMemory -port %s",
+        DYNAMO_LOCAL_DIR_NAME, DYNAMO_LOCAL_DIR_NAME, DYNAMO_LOCAL_SERVER_PORT));
+    Thread.sleep(1000); // wait for start
+    dynamo = DynamoDbClient.builder()
+        .endpointOverride(URI.create(String.format("http://0.0.0.0:%d/", DYNAMO_LOCAL_SERVER_PORT)))
+        .region(Region.US_EAST_1) // dummy region
+        .credentialsProvider(StaticCredentialsProvider.create(
+            AwsBasicCredentials.create("key", "secret"))) // dummy credential
+        .httpClient(UrlConnectionHttpClient.create())
+        .build();
+    properties = new AwsProperties();
+    properties.setGlueCatalogLockEnabled(true);
+    properties.setGlueCatalogLockTable(TABLE_NAME);
+    lockManager = new DynamoLockManager(dynamo, properties);
+  }
+
+  @After
+  public void after() throws Exception {
+    dynamo.deleteTable(DeleteTableRequest.builder()
+        .tableName(TABLE_NAME)
+        .build());
+    dynamoLocalProcess.destroy();
+    dynamoLocalProcess.waitFor();
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    Runtime.getRuntime().exec(String.format("rm %s.tar.gz", DYNAMO_LOCAL_DIR_NAME)).waitFor();
+    Runtime.getRuntime().exec(String.format("rm -rf %s", DYNAMO_LOCAL_DIR_NAME)).waitFor();
+  }
+
+  @Test
+  public void testTableCreation() {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(TABLE_NAME)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when describing DynamoDB lock table");
+    }
+
+  }
+
+  @Test
+  public void testLock_singleProcess() {
+    boolean succeed = lockManager.tryLock("database.table", AwsProperties.GLUE_CATALOG_LOCK_EXPIRE_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s("database.table").build());
+    try {
+      dynamo.getItem(GetItemRequest.builder()
+          .tableName(TABLE_NAME)
+          .key(key)
+          .build());
```

Comment:
If the lock stored the expiration interval, then this could assert that a result was received and that the interval is the one expected.